### DISCUSSION TITLE
Renovate don't commit untracked/changed files in "non vendor" folder for composer manager

### DISCUSSION THREAD

**Discussion by paulo-dc** [renovatebot/renovate#32517] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab, latest

### Please tell us more about your question or problem

Hello, according to a previous [discussion](https://github.com/renovatebot/renovate/discussions/32108), I am now able to run renovate on a wordpress based project and keep up to date, with the composer manager, plugins not installed in a specific folder `wp-content/plugins` (not in `vendor`).

But there is something strange, unlike plugins that are in the `vendor` directory, for the ones that are in the specific folder, the commit with the PR don't have any untracked/changed files. Only deleted ones.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
It's due to the vendor dir being hardcoded for added/modified files. See https://github.com/renovatebot/renovate/blob/894250d392ba025e870134394b0c113f0fe5eb9e/lib/modules/manager/composer/artifacts.ts#L219-L237

This would require either:
- Loosen Renovate to commit _any_ changed file by default (a bit risky), or
- Add a post-update option to opt into committing any changed file (not risky), or
- Add logic for Renovate to understand/parse the extras part to work out the alternative dir

**Reply by paulo-dc** [reply]
Many thanks Rhys for your answer.

All those 3 solutions need to be implemented in a new release of Renovate.

Do you consider that as a bug or a feature request ? Is it something that could be added in the short or medium term?

BR

**Reply by rarkins** [reply]
Feature request. Unlikely to be done unless sponsored

**Reply by paulo-dc** [reply]
Can I create a feature Request from this discussion please ?

**Reply by rarkins** [reply]
Please create a reproduction repo demonstrating the problem + new "Suggest an idea" discussion and we'll convert it to an issue. Put the full information in the discussion (don't say "for full details see ..." linking back to this or the older discussion).
